### PR TITLE
Add `SigningProfile` bootsrappable resource

### PR DESCRIPTION
**Pull request by A-Hilaly**
Description of changes:

This patch adds `SigningProfile` bootstrappable resource to acktest
library. Note that signing profiles cannot be deleted. They can only
be canceled during the clean-up phase. They are automatically
deleted after two years.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/acktest/bootstrapping/dynamodb.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def dynamodb_resource(self):
     def bootstrap(self):
         """Creates a Dynamodb table with an auto-generated name.
         """
-        self.name = resources.random_suffix_name(self.name_prefix, 63)
+        self.name = resources.random_suffix_name(self.name_prefix, 8)
```
Comment (vijtrip2) [inline src/acktest/bootstrapping/dynamodb.py]: You can easily blow past 8 character limit with a table name from tests. Why did you need to change this?

**Review by vijtrip2 — APPROVED** [review]

File: src/acktest/bootstrapping/dynamodb.py (hunk 1 above)
Comment (A-Hilaly) [reply src/acktest/bootstrapping/dynamodb.py]: Modified the wrong file :) just force pushed

Comment (A-Hilaly) [reply src/acktest/bootstrapping/dynamodb.py]: Signing profile names cannot go over 64characters

**Comment by vijtrip2** [comment]
>They can only be canceled during the clean-up phase. They are automatically deleted after two years.

Argh... hopefully it does not cause e2e resources to pileup and then fail due to exceeded limits. But i guess we can do nothing about it.

File: src/acktest/bootstrapping/dynamodb.py (hunk 1 above)
Comment (vijtrip2) [reply src/acktest/bootstrapping/dynamodb.py]: why not make it something like 32? instead of 8.

IIRC, this method will only add suffix if name_prefix is less than 8 chars which is very less IMO. Consider 32 if you don't mind

Comment (A-Hilaly) [reply src/acktest/bootstrapping/dynamodb.py]: Agreed

**Review by vijtrip2 — COMMENTED** [review]
/lgtm
